test/integration/targets/unarchive/tasks/main.yml

```diff
@@ -34,6 +34,26 @@
   pkgng: name=zip,unzip state=present
   when: ansible_pkg_mgr  ==  'pkgng'
 
+- name: check that zip is installed
+  shell: zip --help
+  register: zip_is_installed
+  ignore_errors: yes
+
+- name: check that unzip has option -P (pass password from cli)
+  shell: unzip -P
+  register: unzip_has_modifier_P
+  ignore_errors: yes
+
+- name: check that pexpect is installed
+  shell: python -c "import pexpect"
+  register: pexpect_is_installed
+  ignore_errors: yes
+
+# Run tests only in case we have unzip -P (password modifier) and zip is installed
+- name: Run tests with protected zip
+  include: protected_zip_tests.yml
+  when:   pexpect_is_installed.failed != true and 'zip_is_installed.failed != true' and "'must give decryption password with -P' in unzip_has_modifier_P.stderir"
```

Comment:
Use `zip_is_installed is not failed` instead of `zip_is_installed.failed != true`.

There is typo in `unzip_has_modifier_P.stderr`.